container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java

```diff
@@ -751,6 +751,27 @@ private Inspector maybeConvertData(Inspector data) {
         }
 
         private void renderInspector(Inspector data) throws IOException {
+            if(data.type().equals(Type.ARRAY)) {
+                int entries = data.entryCount();
+                for (int i = 0; i < entries; i++) {
+                    if(!data.entry(i).type().equals(Type.STRING)) {
+                        renderInspectorDirect(maybeConvertData(data));
+                        return;
+                    }
+                }
+
+                generator.writeStartArray();
+                for (int i = 0; i < entries; i++) {
+                    byte[] utf8 = data.entry(i).asUtf8();
+                    try {
+                        generator.writeUTF8String(utf8, 0, utf8.length);
+                    } catch (IOException ex) {
+                        throw new RuntimeException(ex);
```

Comment:
You may want to leave a comment here if exception should never happen. If not ensure that `RuntimeException` is handled correctly. Other code paths wraps this in `UncheckedIOException`. You can likely just not catch is as `renderInspector` already throws `IOException`

Reply:
Thanks, completely missed this from my PoC implementation